test/gurka/test_ferry_connections.cc

```diff
@@ -638,6 +638,78 @@ TEST(Standalone, ConsiderBlockedRoads) {
   EXPECT_EQ(std::get<1>(blocked)->classification(), valhalla::baldr::RoadClass::kServiceOther);
 }
 
+TEST(Standalone, MultiModalReclassify) {
+  // Sometimes ferries are connected by dedicated roads for different modes of transport.
+  // Reclassification should reclassify as much roads as needed to cover all modes of transport.
+
+  const std::string ascii_map = R"(
+      A=====X
+      |
+      |
+      B   I---M==N
+     /|   |\
+    / |   | \
+   E  |   |  K
+   |  |   |  |
+   F  |   |  J
+    \ |   | /
+     \|   |/
+      C   H
+      |   |
+      D---G
+  )";
```

Comment:
What happened in such cases before this PR - reclassification was able to find ABCDGHI shortest path with `path_access == 0` and it stopped there, unable to find better path.